src/setuptools_scm/__init__.py

```diff
@@ -138,62 +52,23 @@ def get_version(
     search_parent_directories: bool = False,
 ) -> str:
     """
-    If supplied, relative_to should be a file from which root may
-    be resolved. Typically called by a script or module that is not
-    in the root of the repository to direct setuptools_scm to the
-    root of the repository by supplying ``__file__``.
-    """
-    version_cls = _validate_version_cls(version_cls, normalize)
-    del normalize
-    if isinstance(tag_regex, str):
-        if tag_regex == "":
-            warnings.warn(
-                DeprecationWarning(
-                    "empty regex for tag regex is invalid, using default"
-                )
-            )
-            tag_regex = DEFAULT_TAG_REGEX
-        else:
-            tag_regex = re.compile(tag_regex)
-    config = Configuration(**locals())
-    maybe_version = _get_version(config)
-
-    if maybe_version is None:
-        _version_missing(config)
-    return maybe_version
+    soft deprecated helper to obtain the vcs version
+    its misused at runtime in numerous projects and has to stay for now
 
+    a replacement supporting editable installation in hatch/setuptools is planned
+    """
 
-def _get_version(config: Configuration) -> str | None:
-    parsed_version = _do_parse(config)
-    if parsed_version is None:
-        return None
-    version_string = _format_version(
-        parsed_version,
-        version_scheme=config.version_scheme,
-        local_scheme=config.local_scheme,
-    )
-    if config.write_to is not None:
-        dump_version(
```

Comment:
my version file is no longer being dumped anywhere and I am trying to figure out why.